backend/workflow_manager/file_execution/internal_views.py

```
@@ -29,111 +30,57 @@ class FileExecutionInternalViewSet(viewsets.ModelViewSet):
+        This avoids the expensive filtered queryset (3-table JOIN + query params)
+        for single-object operations (retrieve, update, destroy).
+        """
+        pk = self.kwargs.get(self.lookup_field)
+        queryset = WorkflowFileExecution.objects.filter(id=pk)
         queryset = filter_queryset_by_organization(
             queryset, self.request, "workflow_execution__workflow__organization"
         )
+        try:
+            obj = queryset.get()
+        except WorkflowFileExecution.DoesNotExist:
+            raise NotFound(f"No file execution found with ID {pk}")
```

Ruff (0.15.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)